src/clients.rs

```diff
@@ -351,17 +351,23 @@ impl StateClient for StateStore {
         let staking_address = bs58::encode(staking_address).into_string();
         println!("Staking address: {}", staking_address);
 
-        let stake = res.ok_or(Error::NotStaked).map(
-            |StakeData {
-                 amount,
-                 reward,
-                 counter,
-             }| StakeInfo {
-                amount,
-                reward,
-                counter,
-            },
-        )?;
+        // FIX_ME: proper solution should to return an Option<StakeInfo>
+        // changing the trait implementation. That would reflect the state of
+        // the stake contract. It would be up to the consumer to decide what to
+        // do with a None
+        let stake = res
```

Comment:
The internal wallet_core `allow` function is returning error if `fetch_stake` returns error. That's the culprit, hence I'm removing the error for NotStaked from `fetch_stake`.

Please check if this PR solve your issue

Reply:
Ps: no one should return error if requesting stake info for a wallet that has no stake. It will say "No active stake for this address, and no reward". That's it